### PR TITLE
Modern theme

### PR DESCRIPTION
**Pull request by Blackbaud-MattGregg**
Adding modern style for sectioned form.

Updates to vertical tabs modern style
  - grouping header font bold on selected state
  - spacing in tabs and content section
  - removed focused border on mouse click
  - switched expand/collapse arrows back to up/down for accordions

### CODE REVIEW

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme)

File: src/app/docs/sectioned-form/demo/demo-address-form.component.html
Comment (Blackbaud-TrevorBurch) [inline src/app/docs/sectioned-form/demo/demo-address-form.component.html:1]: I know these are just for visual testing and dev use - but is there a reason we are gutting these?

Comment (Blackbaud-TrevorBurch) [reply src/app/docs/sectioned-form/demo/demo-address-form.component.html:1]: Actually - these first two are the actual demo that shows up for consumers on the docs - it doesn't seem like we would want to gut those.

File: src/app/public/modules/vertical-tabset/vertical-tabset.component.scss
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/vertical-tabset/vertical-tabset.component.scss:25]: Have we verified this style working? I could be wrong but I wouldn't have expected this style to work in this order. I would have thought this inner style would have had to go in the `@include sky-host-responsive-container-sm-min()` below (currently on line 63) due to the ordering of the css classes on the page.

File: src/app/visual/sectioned-form/sectioned-form-visual.component.ts
Comment (Blackbaud-TrevorBurch) [inline src/app/visual/sectioned-form/sectioned-form-visual.component.ts:16]: There are some odd spacing things going on in this block. Can we bump the closing constructor parenthesis onto a new line and unindent the `themeSettingsChange` declaration?

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html:26]: The `ngIf` should go on he same line as the `<div`

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html:42]: `[tabIndex]="-1"` should be before the `(click)="toggleMenuOpen()"`

Our standard ordering is 

```
<component-name *structuralDirective=""
  nativeattributes=""
  unboundProperty=""
  [inputs]=""
  [(twoWay)]=""
  (outputs)=""
  #templateReferenceVariables
>
  Some content here.
</component-name>
```

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html:33]: The `*skyThemeIf` should go on the same line as the `<div`

File: e2e/sectioned-form.e2e-spec.ts
Comment (Blackbaud-TrevorBurch) [inline e2e/sectioned-form.e2e-spec.ts:12]: Nitpicky - but could we move this import statement to be directly below the other `@skyux-sdk/e2e` import? That way they are together when looking at the imports?

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html:15]: I'm curious about these chevron changes from a design standpoint. 

The change from up-down to right-down was intentionally made when we converted this to modern theme. This seems to be undoing that.

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-TrevorBurch) [inline src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html:3]: I'm curious in why we are adding this `skyThemeIf` here and then changing how the `skyThemeIf`s on the chevron and modern chevron equivalent are working? Unless I'm missing something we are now achieving the same thing with duplicated code of this `div` down below.

**Review by Blackbaud-TrevorBurch — CHANGES_REQUESTED** [review]
@Blackbaud-MattGregg - Just a few minor things. And we should probably get Todd or Adam to look at this too just to verify the changes from a design team standpoint. 

There are definitely changes here that would require the vertical tabs screenshots to be replaced - but I would defer to the design team on if those are valid or not.

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html:42]: NOTE: I see now you just copied from above - and what was there was just already wrong :)

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-TrevorBurch) [reply src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html:15]: Also a note that if we are changing the chevron directions back then I don't think we need the `skyThemeIf` parts at all as we would then just have functionally the same things and should just be able to use the `sky-chevron` for both themes. We only had the `skyThemeIf` stuff because the `sky-chevron` didn't support the deliberate move we made to change the direction to `right` in modern theme.

File: src/app/docs/sectioned-form/demo/demo-address-form.component.html
Comment (Blackbaud-MattGregg) [reply src/app/docs/sectioned-form/demo/demo-address-form.component.html:1]: @Blackbaud-TrevorBurch Gutting them b/c the content isn't representative of what it should be (i.e., a form) and weren't formatted using SKY UX components (i.e., not how one should display that kind of content). This updates moves them more to the approach for docs used in Tabs and Vertical tabs - a label just stating what would display here instead of fake content.

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-MattGregg) [reply src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html:3]: @Blackbaud-TrevorBurch It seemed like I needed to add the `skyThemeif` at that level because the ngclass on that div needed to be different btw modern and default. This wasn't done on the initial styles updates. The modern version uses a different font class (sky-font-emphasized vs font-emphasized) to achieve a bolder font weight in modern style. Maybe there is a better way to do this? I just did it at the top level div b/c there was this change the chevrons were different.

File: src/app/public/modules/vertical-tabset/vertical-tabset.component.scss
Comment (Blackbaud-MattGregg) [reply src/app/public/modules/vertical-tabset/vertical-tabset.component.scss:25]: Yes. it was working, but you're right it should probably be at bottom to make it consistent that modern styles are down there together. I moved it and confirmed it worked. Also found I needed to add this to section form scss as it wasn't getting this from this vertical tab declaration. thx.

File: src/app/visual/sectioned-form/sectioned-form-visual.component.ts
Comment (Blackbaud-MattGregg) [reply src/app/visual/sectioned-form/sectioned-form-visual.component.ts:16]: Addressed in recently commit. https://github.com/blackbaud/skyux-tabs/pull/178/commits/f23bd3b82e0f0bcaa9face16e77ee5d8720a483d

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-MattGregg) [reply src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html]: Think I addressed this in recent commit. https://github.com/blackbaud/skyux-tabs/pull/178/commits/f23bd3b82e0f0bcaa9face16e77ee5d8720a483d

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-MattGregg) [reply src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html]: Addressed in recent commit. https://github.com/blackbaud/skyux-tabs/pull/178/commits/f23bd3b82e0f0bcaa9face16e77ee5d8720a483d

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-MattGregg) [reply src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html]: Addressed in recent commit. https://github.com/blackbaud/skyux-tabs/pull/178/commits/f23bd3b82e0f0bcaa9face16e77ee5d8720a483d

File: e2e/sectioned-form.e2e-spec.ts
Comment (Blackbaud-MattGregg) [reply e2e/sectioned-form.e2e-spec.ts]: yep. Rearranged these in recent commit. https://github.com/blackbaud/skyux-tabs/commit/f23bd3b82e0f0bcaa9face16e77ee5d8720a483d

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-MattGregg) [reply src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html:14]: @Blackbaud-TrevorBurch Gotcha on the `sky-chevron`. I switched this back. It just means there wasn't a way to make the selected group's arrow the selected color but not a biggie. 
I still kept the `*skyThemei`f sections b/c the `<div>` needs a different font class when the sub menu is open in modern. If there is a way to use` [SkyThemeClass]` with that dynamic state `: subMenuOpen()` then I could just have the classes change based on the theme instead of the whole `div`.

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-MattGregg) [reply src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html]: Removed the duplicated div and added a modern class (sky-font-emphasized) to the subMenuOpen state to achieve the modern font weight instead.

File: src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html
Comment (Blackbaud-MattGregg) [reply src/app/public/modules/vertical-tabset/vertical-tabset-group.component.html:16]: Updated to just have `sky-chevron` for both themes. Switching back to up/down arrow for representing expanded/collapsed for the accordion control.

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme)

**Review by Blackbaud-MattGregg — COMMENTED** [review]
@Blackbaud-TrevorBurch I think I have addressed all of these in recent commits.

**Review by Blackbaud-ToddRoberts — CHANGES_REQUESTED** [review]
- Disabled state needs `sky-font-deemphasized` for modern so the font size doesn't get reduced. Should it also be italicized to differentiate from enabled-not-active?
- On click, text is shifting to deemph color but I think it should stay default color

**Comment by Blackbaud-MattGregg** [comment]
@Blackbaud-ToddRoberts Updated disabled states for font size problem and added italicized. I don't think we'd really specified that anywhere but agree it helps to differentiate esp. in modern visual style. 

I'm not sure what text you saw shifting to deemph color onClick. I couldn't find that. Could you point out where that was?

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme)

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme)

**Comment by Blackbaud-MattGregg** [comment]
@Blackbaud-ToddRoberts Found the text shifting color. Somehow missed that the first time. I believe your requests are now included.

**Comment by blackbaud-johnly** [comment]
@Blackbaud-MattGregg, should the demo changes you made to the content of the *Addresses* and *Phone numbers* tabs also be reflected in the Stackblitz demos? Right now, those demos still include the content that you removed from the demo.

**Review by Blackbaud-ToddRoberts — APPROVED** [review]

**Review by Blackbaud-TrevorBurch — DISMISSED** [review]

**Comment by Blackbaud-MattGregg** [comment]
@blackbaud-johnly If they should be kept in sync, happy to do that. It's not something I've done before. Any pointers on procedure here? It doesn't look like they are in a repo but maybe on StackBlitz itself?

**Comment by Blackbaud-TrevorBurch** [comment]
@Blackbaud-MattGregg the code examples are used for StackBlitz. They are located at `src/app/public/plugin-resources/code-examples/sectioned-form`

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme)

**Comment by Blackbaud-MattGregg** [comment]
Thanks @Blackbaud-TrevorBurch I've updated the examples in StackBlitz.  I requested you re-review as it showed github dismissed stale reviews with my last commit I pushed. Does Alex need to review this as well?

**Comment by blackbaud-johnly** [comment]
> @blackbaud-johnly If they should be kept in sync, happy to do that. It's not something I've done before. Any pointers on procedure here? It doesn't look like they are in a repo but maybe on StackBlitz itself?

Sorry for the slow follow-up, @Blackbaud-MattGregg. Looks like you figured it out before I could get back to you.

**Review by Blackbaud-TrevorBurch — APPROVED** [review]

**Comment by blackbaud-ado** [comment]
Preview build at [https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme](https://developer.blackbaud.com/skyux-tabs?_sp.spa=pr-Modern-theme)

**Pull request merged**
